### PR TITLE
remove calico-upgrade and update resources

### PR DESCRIPTION
**Pull request by kwmonroe**
All supported CK releases use calico v3.  This PR drops the v2->v3 migration tools from the canal charm.

Fixes https://bugs.launchpad.net/charm-canal/+bug/1960970

### CODE REVIEW

**Comment by kwmonroe** [comment]
There's an ugly piece in here:  calico versions for the canal and calico charms have diverged in CK 1.24.  We were relying on a pinned [build-calico-resource.sh](https://github.com/charmed-kubernetes/layer-calico/blob/5dab860987023b9b2570870b0f37d958bb7fd9b2/build-calico-resource.sh) from the calico repo to build calico resources for the canal charm, but that is no longer possible after the charmhub migration.

Instead, I took relevant bits from that script and dropped them into canal's `build-canal-resource.sh` script.  It's ugly, and it means we build calico resources one way and flannel another.  Open to suggestions to make this cleaner.

File: DEVELOPING.md
Comment (addyess) [inline DEVELOPING.md:55]: this bundle `canonical-kubernetes-canal` doesn't exist. 

I think you'd have to do `charmed-kubernetes` + some canal overlay

File: tests/data/bundle.yaml
Comment (addyess) [inline tests/data/bundle.yaml:13]: i've been pushing these to `latest/edge` for when we begin to use the other tracks like `1.23/edge`.  I don't know if that's actually necessary -- but at least its more explicit

**Review by addyess — COMMENTED** [review]
i don't quit understand the reasoning on the removal of `calico-upgrade`.  I might need more time on that.  But provided some info on the README and integration tests setup.  

The rest of it looks like it will solidly produce a build -- i just can't comment on whether it will function or not without more time

File: DEVELOPING.md
Comment (kwmonroe) [reply DEVELOPING.md:55]: You are correct that we would normally want c-k to be the "real" bundle, modified as needed with overlays.

However, there is a c-k-c bundle in ch. It made it over during the sync because it was a promulgated bundle in cs:

https://charmhub.io/canonical-kubernetes-canal

Comment (addyess) [reply DEVELOPING.md:55]: 🤮

File: reactive/calico.py
Comment (Cynerva) [inline reactive/calico.py:104]: I think calico-v3-data-ready could be removed too. It was used to stop certain handlers like `install_calico_binaries` from running until after v2 -> v3 data migration had occurred. It serves no function now.

**Review by Cynerva — APPROVED** [review]
Looks good to me, thank you. I have no concerns with removing calico-upgrade, and the approach you've taken here looks safe to me.

I'll leave it up to you if you want to address my comment about calico-v3-data-ready before merging this or not.

Please also take a look at https://github.com/charmed-kubernetes/jenkins/pull/818 which I believe will be needed for CI to accept this change.

File: reactive/calico.py
Comment (kwmonroe) [reply reactive/calico.py]: Excellent suggestion. Might even see a perf boost when we stop running the `v3_data_ready` handler every 5 minutes :)

**Comment by kwmonroe** [comment]
> i don't quit understand the reasoning on the removal of `calico-upgrade`

Thanks for the review. The reasoning for this removal is 2-fold:

1. calico-v2 hasn't been the default since CK 1.16; the last supported release that would have entertained the idea of migrating calico v2->v3 would have been CK 1.18.  Since we're well past that, drop the superfluous resource/handler code.
2. we [can't build](https://github.com/projectcalico/calico-upgrade/issues/42) the `calico-upgrade-arm64` resource, so we had been fetching that from the calico charm. When calico moved to charmhub, it went with v3.21.x. Canal needs v3.10.x. There is no charmhub calico charm with a v3.10.x `calico-upgrade-arm64` resource that we can pull from. Thankfully, we don't need one (see point 1).

**Pull request merged**
